lib/dispatcher.rb

```diff
@@ -7,7 +7,18 @@ def initialize(message)
   end
 
   def dispatch!
-    job_class.perform_later(endpoint, timestamp, category, message)
+    Rails.logger.info("Dispatching: #{@message.to_h}")
+
+    relevant_filters = Filter.matching_message(@message).preload(:endpoints)
+
+    endpoints = relevant_filters.map(&:endpoints).flatten.uniq
```

Comment:
It would be nice to be able to do these two lines at the db level, but I'm not sure if it is worth the effort

Reply:
Right now we have a single query in the `relevant_filters = ...` line, and zero queries in the second line.

Actually first line produces two queries because of `preload` statement. I do prefer it over `include`, since it caused a lot of problems in Foreman.

Reply:
What I meant was the `.map.flatten.uniq` part and if it wouldn't be better to have it implemented using some kind of `select distinct` or `select endpoints where id in (...message matching subquery...)` instead of loading everything and doing the transformation in the application.

Reply:
True, but the list _should_ be quite small, so I hope it wouldn't blow up the resources.
The alternative will look something like this:

``` ruby
Endpoint.joins(:endpoint_filters).where(endpoint_filters: { id: Filter.Filter.matching_message(@message).select(:id) }).distinct
```